### ISSUE TITLE
Query widget keeps requesting in v2.1

### ISSUE THREAD

**Issue by xclud** [zino-app/graphql-flutter#387]
I have a Query widget on the page. It keeps requesting to the server with rate of tens of requests per second. After a little while it throws a SocketException (Which i guess is caused by too many open ports).

And it never shows the contents (My case a ListView) but rather showing a progressbar (result.loading is always true).

I have a little server at home for development purposes. My servers goes out of resources because of too many requests/connections.

This bug appeared since i updated to Version 2.1. Version 2.0 works fine (I haven't checked previous versions).

**Comment by mainawycliffe** [comment]
Have you configured polling by any chance under QueryOptions? And could you please share a reproducible example.

**Comment by xclud** [comment]
```
Widget _getContent() {
return Query(
  options: QueryOptions(
	document: r'''
	{
	  events {
		id
		name
		group {
		  id
		  name
		}
	  }
	}
	''',
  ),
  builder: (QueryResult result,
	  {VoidCallback refetch, FetchMore fetchMore}) {
	if (result.errors != null) {
	  return Text(result.errors.toString());
	}

	if (result.loading) {
	  return Center(
		child: const CircularProgressIndicator(),
	  );
	}
	return RefreshIndicator(
		onRefresh: () async {
		  refetch();
		},
		child: _getItems(result.data['events']));
  },
);
}
```

I used ClientProvider from this repository like this:

```
return ClientProvider(
  uri: NetworkHelper.GRAPHQL_ENDPOINT,
  child: MaterialApp(
	debugShowCheckedModeBanner: false,
	title: 'App Name',
	builder: (BuildContext context, Widget child) {
	  return Directionality(
		textDirection: TextDirection.rtl,
		child: child,
	  );
	},
	home: HomePage()
  ),
);
```

**Comment by mainawycliffe** [comment]
Do you mean `GraphQLProvider` instead of `ClientProvider`?

**Comment by xclud** [comment]
Here is the ClientProvider. I took it from /example directory (Or i don't remember?) few weeks ago.

```
import 'package:graphql_flutter/graphql_flutter.dart';
import 'package:flutter/material.dart';

String uuidFromObject(Object object) {
  if (object is Map<String, Object>) {
    final String typeName = object['__typename'] as String;
    final String id = object['id'].toString();
    if (typeName != null && id != null) {
      return <String>[typeName, id].join('/');
    }
  }
  return null;
}

final OptimisticCache cache = OptimisticCache(
  dataIdFromObject: uuidFromObject,
);

ValueNotifier<GraphQLClient> clientFor(
    {@required String uri, String subscriptionUri}) {
  final HttpLink httpLink = HttpLink(
    uri: uri,
  );

  Link link = httpLink;// authLink.concat(httpLink);

  if (subscriptionUri != null) {
    final WebSocketLink websocketLink = WebSocketLink(
      url: subscriptionUri,
      config: SocketClientConfig(
        autoReconnect: true,
        inactivityTimeout: Duration(seconds: 30),
      ),
    );

    link = link.concat(websocketLink);
  }

  return ValueNotifier<GraphQLClient>(
    GraphQLClient(
      cache: cache,
      link: link,
    ),
  );
}

/// Wraps the root application with the `graphql_flutter` client.
/// We use the cache for all state management.
class ClientProvider extends StatelessWidget {
  ClientProvider(
      {@required this.child,
      @required String uri,
      String subscriptionUri})
      : client = clientFor(
          uri: uri,
          subscriptionUri: subscriptionUri,
        );

  final Widget child;
  final ValueNotifier<GraphQLClient> client;

  @override
  Widget build(BuildContext context) {
    return GraphQLProvider(
      client: client,
      child: child,
    );
  }
}

```

**Comment by xclud** [comment]
Sorry for bothering you. It was my fault.

For other people who may encounter my problem:

I was a `stateful` `MyApp` class as in `void main() => runApp(MyApp());` and deciding what to put in `MaterialApp.home` depending on some state value.

It turned out the MyApp class must inherit from `StatelessWidget` always, having state variable or not.

This problem appeared with v2.0 too. And after i fixed the mistake it worked well with v2.0 and v2.1.

I am closing this issue as it was not related to this package.
